src/Octokit.Webhooks/Converter/StringEnumEnumerableConverter.cs

```diff
@@ -0,0 +1,64 @@
+namespace Octokit.Webhooks.Converter;
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Octokit.Webhooks.Extensions;
+
+public sealed class StringEnumEnumerableConverter<TEnum> : JsonConverter<IEnumerable<StringEnum<TEnum>>>
+    where TEnum : struct
+{
+    public override IEnumerable<StringEnum<TEnum>> Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options) =>
+        ReadInternal(ref reader, options);
+
+    public override void Write(Utf8JsonWriter writer, IEnumerable<StringEnum<TEnum>> value, JsonSerializerOptions options) =>
+        WriteInternal(writer, value, options);
+
+    internal static IEnumerable<StringEnum<TEnum>> ReadInternal(ref Utf8JsonReader reader, JsonSerializerOptions options)
+    {
+        if (reader.TokenType == JsonTokenType.Null)
+        {
+            throw new JsonException("Unexpected null value.");
+        }
+
+        var jsonSerializerOptions = new JsonSerializerOptions(options);
```

Comment:
`JsonSerializerOptions` instances are really expensive to creat because they do a bunch of reflection internally to create things when instantiated.

I think you'd want to have a single instance in a static cache somewhere and re-use it between calls.

Reply:
I can create a static instance of `JsonSerializerOptions`, but then I lose any custom configuration that is passed in e.g. `Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)`

Do you have any concerns with this?

Reply:
Maybe try a microbenchmark to see how it handles? Maybe the copy-ish constructor has better perf.

I got burned by some bad DI registration once where instead of a singleton we were creating a new instance per request and a ~1ms/req response time for an API in front of Redis went to ~15ms/req 😅

Reply:
I am not seeing any meaningful difference. 

```
BenchmarkDotNet=v0.13.2, OS=Windows 11 (10.0.22621.900)
Intel Core i9-10980XE CPU 3.00GHz, 1 CPU, 36 logical and 18 physical cores
.NET SDK=6.0.403
  [Host]   : .NET 6.0.11 (6.0.1122.52304), X64 RyuJIT AVX2
  .NET 6.0 : .NET 6.0.11 (6.0.1122.52304), X64 RyuJIT AVX2

Job=.NET 6.0  Runtime=.NET 6.0  

|     Method |     Mean |     Error |    StdDev |
|----------- |---------:|----------:|----------:|
|   NoParams | 1.393 us | 0.0277 us | 0.0447 us |
| CopyParams | 1.420 us | 0.0283 us | 0.0415 us |
```

<details>
<summary>Micro Benchmark</summary>

```C#
[SimpleJob(RuntimeMoniker.Net60)]
public class JsonSerializerOptionsBenchmark
{
    private JsonSerializerOptions? options;

    [GlobalSetup]
    public void Setup() => this.options = new JsonSerializerOptions();

    [Benchmark]
    public JsonSerializerOptions NoParams() => new();

    [Benchmark]
    public JsonSerializerOptions CopyParams() => new(this.options!);
}
```

</details>

Reply:
I'll try and dig up the PR tomorrow and get the numbers. Maybe it was in 5.0 and was improved by something in 6.0.

Reply:
I think [the docs for .NET 7 agree with you][1]. I'll migrate this to a static instance.

I don't _think_ this will cause any issues, as we have full control over deserialization.

[1]: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/configure-options?pivots=dotnet-7-0